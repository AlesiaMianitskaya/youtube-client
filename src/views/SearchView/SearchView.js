export default class SearchView {
  constructor() {
    this.onChange = () => null;

    const form = document.createElement('form');
    form.setAttribute('id', 'search-form');
    form.setAttribute('class', 'search-form');
    document.body.appendChild(form);

    form.innerHTML = `
      <input id="search" class="search-box" type="text" name="search" placeholder="Search Youtube" />
      <input type="submit" class="submit-button" value />
    `;

    form.addEventListener('submit', e => {
      e.preventDefault();
      this.onChange(e.target.elements.search.value);
    });
  }
}
