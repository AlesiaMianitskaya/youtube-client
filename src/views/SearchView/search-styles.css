@import url('https://fonts.googleapis.com/css?family=Roboto:300,400,500');
body {
  font-family: 'Roboto', sans-serif;
  font-size: 20px;
  background-color: #e8e8e8;
}

.search-form {
  width: 600px;
  height: 40px;
  margin: 30px auto;
}

.search-box {
  width: 500px;
  height: 40px;
  padding: 0px 0px 0px 15px;
  border: 2px solid #3c3d3a;
  border-radius: 20px;
  font-size: 1rem;
}

.search-box:focus,
.search-box:hover {
  background-color: #f2d2ea;
  outline: none;
}

.submit-button {
  width: 50px;
  height: 40px;
  margin-left: 10px;
  padding: 0px;
  border: 0px;
  background-image: url('./assets/search.png');
  background-color: #e8e8e8;
  background-repeat: no-repeat;
  background-size: contain;
  vertical-align: middle;
}

.submit-button:focus {
  outline: none;
}

/* 
  ##Device = Tablets, Ipads (portrait)
  ##Screen = from 768px to 1024px
*/
@media (min-width: 768px) and (max-width: 1024px) {
  .search-form {
    width: 450px;
  }
  .search-box {
    width: 350px;
  }
}

/* 
  ##Device = Low Resolution Tablets, Mobiles
  ##Screen = from 481px to 767px
*/
@media (min-width: 481px) and (max-width: 767px) {
  .search-form {
    width: 370px;
  }
  .search-box {
    width: 280px;
  }
}

/* 
  ##Device = Most of the Smartphones Mobiles
  ##Screen = from 320px to 479px
*/
@media (min-width: 320px) and (max-width: 480px) {
  .search-form {
    width: 310px;
    height: 30px;
  }
  .search-box {
    width: 220px;
    height: 30px;
  }

  .submit-button {
    width: 40px;
    height: 30px;
  }
}
