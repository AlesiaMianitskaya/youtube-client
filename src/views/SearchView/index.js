import SearchView from './SearchView';

import './search-styles.css';

export default SearchView;
