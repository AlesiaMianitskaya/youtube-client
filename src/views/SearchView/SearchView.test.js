import SearchView from './SearchView';

describe('SearchView.prototype.constructor', () => {
  it('Should be an instance of Function', () => {
    expect(SearchView.prototype.constructor).toBeInstanceOf(Function);
  });

  it('Should be render search box correctly', () => {
    const searchBox = new SearchView();
    expect(searchBox).toBeInstanceOf(SearchView);
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
