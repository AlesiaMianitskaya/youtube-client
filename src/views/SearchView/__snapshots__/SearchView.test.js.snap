// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchView.prototype.constructor Should be render search box correctly 1`] = `
"<form id=\\"search-form\\" class=\\"search-form\\">
      <input id=\\"search\\" class=\\"search-box\\" type=\\"text\\" name=\\"search\\" placeholder=\\"Search Youtube\\">
      <input type=\\"submit\\" class=\\"submit-button\\" value=\\"\\">
    </form>"
`;
